app/_src/gateway/upgrade/dual-cluster.md

```diff
@@ -0,0 +1,96 @@
+mode deployments and for control planes in hybrid mode. 
+
+With a dual-cluster upgrade, you deploy a new cluster of version Y alongside the current version X, 
+so that two clusters serve requests concurrently during the upgrade process. 
+As you add nodes, you will gradually adjust the traffic ratio between the two clusters to 
+switch traffic over from the old cluster to the new one.
+
+![Dual-cluster upgrade workflow](/assets/images/products/gateway/upgrade/dual-cluster-upgrade.png)
+> _Figure 1: Dual-cluster upgrade workflow_
+
+This method has limitations on automatically generated runtime metrics that rely on the database. 
+For example, if the Rate-Limiting-Advanced (RLA) plugin is configured to store request counters in 
+the database, the counters between database X and database Y are not synchronized. 
+The impact scope depends on the `window_size` parameter of the plugin. 
+{% if_version eq:3.4.x %}
+Similarly, the same limitation applies to Vitals if you have a large amount of buffered metrics in 
+PostgreSQL.
+{% endif_version %}
+{% if_version lte:3.3.x %}
+Similarly, the same limitation applies to Vitals if you have a large amount of buffered metrics in 
+PostgreSQL or Cassandra.
+{% endif_version %}
+
+## Prerequisites
+
+* Review the [general upgrade guide](/gateway/{{page.kong_version}}/upgrade/) to prepare for the upgrade and review your options.
+* You have a traditional deployment or you need to upgrade the control planes (CPs) in a hybrid mode deployment.
+* You have enough resources to temporarily run an additional {{site.base_gateway}} cluster alongside your existing cluster.
+
+## Upgrade using the dual-cluster method
+
+This guide refers to the old version as cluster X and the new version as cluster Y.
+
+{:.note}
+> The following steps are intended as a guideline.
+The exact execution of these steps will vary depending on your environment. 
+
+1. Stop any {{site.base_gateway}} configuration updates (e.g. Admin API calls). 
+This is critical to guarantee data consistency between cluster X and cluster Y.
+
+    To keep data consistency between the two clusters, you must not execute any write operations through Admin API, Kong Manager, or direct database updates. 
+    This upgrade strategy is the safest of all available strategies and ensures that there is no planned business downtime during the upgrade process.
+
+2. Back up data from the current cluster Y by following the 
+[Backup guide](/gateway/{{page.kong_version}}/upgrade/backup-and-restore/).
+
+3. Evaluate factors that may impact the upgrade, as described in [Upgrade considerations](/gateway/{{page.kong_version}}/upgrade/#preparation-upgrade-considerations/).
+You may have to consider customization of both `kong.conf` and {{site.base_gateway}} configuration data.
+
+4. Evaluate any [breaking changes](/gateway/{{page.kong_version}}/breaking-changes/) that may 
```

Comment:
Not only breaking changes, but also the complete changelog: https://docs.konghq.com/gateway/changelog/